sktime/annotation/datagen.py

```diff
@@ -241,6 +241,53 @@ def label_piecewise_normal(
     return np.repeat(unique_labels, lengths)
 
 
+def piecewise_poisson(
+    frequency: npt.ArrayLike,
+    lengths: npt.ArrayLike,
+    random_state: Union[int, np.random.RandomState] = None,
+) -> npt.ArrayLike:
+    """
+    Generate series using Possion distribution.
+
+    Each segment has length specified in ``lengths`` and data sampled from a Poisson
+    distribution with expected frequency from ``frequency``.
+
+    Parameters
+    ----------
+    frequency : array_like
+        Expected frequency of the segments to be generated
```

Comment:
Small change - but I would recommend you call this "lamba" rather than "frequency" (just to follow the typical syntax seen with Poisson distributions) and explain that it is the expected number (and variance! Aren't Poisson distributions cool?) of events within each time interval/data point.

Reply:
This is one point where python and maths don't play well since `lambda` is a reserved keyword in python. I've been in this situation before and still don't have a good solution for an alternative name to something everyone knows should be called `lambda`.

Reply:
Oh whoops, of course!  Perhaps in this case `lambdas` would work (since it is an array).  And then could use `lam` or similar when iterating over it.